### ISSUE TITLE
Java8FIPS documentation tests failed to initialize the SSLContext

### ISSUE THREAD

**Issue by polyfractal** [elastic/elasticsearch#49837]
Logs were truncated unfortunately, but it looks like the `java8fips` documentation tests failed due to a FIPS error?  This was on 7.x

```
Caused by: java.security.KeyManagementException: FIPS mode: only SunJSSE TrustManagers may be used
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/368/

https://gradle-enterprise.elastic.co/s/4jcn7y3ozwyhy/console-log?task=:docs:integTestRunner

```
12:16:56 » ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [node-0] uncaught exception in thread [main]
12:16:56 »  org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: failed to load plugin class [org.elasticsearch.xpack.core.XPackPlugin]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:174) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125) ~[elasticsearch-cli-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.cli.Command.main(Command.java:90) ~[elasticsearch-cli-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  Caused by: java.lang.IllegalStateException: failed to load plugin class [org.elasticsearch.xpack.core.XPackPlugin]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:614) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundle(PluginsService.java:556) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:471) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:309) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	... 6 more
12:16:56 »  Caused by: java.lang.reflect.InvocationTargetException
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
12:16:56 »  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
12:16:56 »  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:605) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundle(PluginsService.java:556) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:471) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:309) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	... 6 more
12:16:56 »  Caused by: org.elasticsearch.ElasticsearchSecurityException: failed to load SSL configuration [xpack.security.http.ssl]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadConfiguration(SSLService.java:526) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.lambda$loadSSLConfigurations$4(SSLService.java:499) ~[?:?]
12:16:56 »  	at java.util.HashMap.forEach(HashMap.java:1289) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadSSLConfigurations(SSLService.java:499) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.<init>(SSLService.java:140) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.XPackPlugin.<init>(XPackPlugin.java:142) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
12:16:56 »  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
12:16:56 »  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:605) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundle(PluginsService.java:556) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:471) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:309) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	... 6 more
12:16:56 »  Caused by: org.elasticsearch.ElasticsearchException: failed to initialize the SSLContext
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.createSslContext(SSLService.java:455) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.createSslContext(SSLService.java:431) ~[?:?]
12:16:56 »  	at java.util.HashMap.computeIfAbsent(HashMap.java:1127) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadConfiguration(SSLService.java:523) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.lambda$loadSSLConfigurations$4(SSLService.java:499) ~[?:?]
12:16:56 »  	at java.util.HashMap.forEach(HashMap.java:1289) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadSSLConfigurations(SSLService.java:499) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.<init>(SSLService.java:140) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.XPackPlugin.<init>(XPackPlugin.java:142) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
12:16:56 »  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
12:16:56 »  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:605) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundle(PluginsService.java:556) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:471) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:309) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	... 6 more
12:16:56 »  Caused by: java.security.KeyManagementException: FIPS mode: only SunJSSE TrustManagers may be used
12:16:56 »  	at sun.security.ssl.SSLContextImpl.chooseTrustManager(SSLContextImpl.java:120) ~[?:?]
12:16:56 »  	at sun.security.ssl.SSLContextImpl.engineInit(SSLContextImpl.java:83) ~[?:?]
12:16:56 »  	at javax.net.ssl.SSLContext.init(SSLContext.java:282) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.createSslContext(SSLService.java:448) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.createSslContext(SSLService.java:431) ~[?:?]
12:16:56 »  	at java.util.HashMap.computeIfAbsent(HashMap.java:1127) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadConfiguration(SSLService.java:523) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.lambda$loadSSLConfigurations$4(SSLService.java:499) ~[?:?]
12:16:56 »  	at java.util.HashMap.forEach(HashMap.java:1289) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.loadSSLConfigurations(SSLService.java:499) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.ssl.SSLService.<init>(SSLService.java:140) ~[?:?]
12:16:56 »  	at org.elasticsearch.xpack.core.XPackPlugin.<init>(XPackPlugin.java:142) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
12:16:56 »  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
12:16:56 »  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
12:16:56 »  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_231]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadPlugin(PluginsService.java:605) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundle(PluginsService.java:556) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.loadBundles(PluginsService.java:471) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:163) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:309) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.node.Node.<init>(Node.java:253) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:221) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:349) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170) ~[elasticsearch-7.6.0-SNAPSHOT.jar:7.6.0-SNAPSHOT]
12:16:56 »  	... 6 more
```

**Comment by jkakavas** [comment]
This was introduced in e6f530c1671, I will take a look

**Comment by jkakavas** [comment]
So, we now (after https://github.com/elastic/elasticsearch/commit/e6f530c1671dba09ac487f63fa4ef2f836093fe2) wrap the TrustManager with our TLSDiagnosticTrustManager and this is not allowed in FIPS mode when using the SunJSSE. We had discussed with @tvernum and he was planning to set `xpack.security.ssl.diagnose.trust` false when `fips.mode.enabled` is true so we should follow up on that, I'll open an issue to keep track

**Comment by jkakavas** [comment]
Raised https://github.com/elastic/elasticsearch/pull/49888 but this won't fix the test failures. We need to set `"xpack.security.fips_mode.enabled"` in all of our nodes in integTest ( we probably should have been doing that already ) - I'll raise an additional build/CI PR soon

**Comment by ebadyano** [comment]
another one:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk13,ES_RUNTIME_JAVA=java8fips,nodes=general-purpose/426/console

**Comment by jkakavas** [comment]
This was resolved by  #49900
